sdk/framework/openapi.go

```diff
@@ -534,31 +530,624 @@ func documentPath(p *Path, specialPaths *logical.Paths, requestResponsePrefix st
 	return nil
 }
 
-// constructRequestResponseName joins the given path with prefix & suffix into
-// a CamelCase request or response name.
+// When constructing a request or response name from a prefix + path, certain
+// paths result in very long names or names with duplicate parts. For such paths,
+// we use custom path mappings instead to generate request/response names.
+type knownPathKey struct {
+	mount string
+	path  string
+}
+
+type knownPathValue struct {
+	name             string
+	operationImplied bool // if true, operation will not be appended to the resulting name
+}
+
+var knownPathMappings = map[knownPathKey]knownPathValue{
```

Comment:
are these _all_ the registered endpoints or just the ones that the default resolved `operationId` needs to be changed?

Reply:
Currently, all the known endpoints are specified in this map. If new ones are added the would get a sensible default but would yield a better name if added to the map.